library/ssl_tls12_server.c

```diff
@@ -1203,18 +1203,6 @@ static int ssl_parse_client_hello( mbedtls_ssl_context *ssl )
     MBEDTLS_SSL_DEBUG_MSG( 3, ( "client hello, protocol version: [%d:%d]",
                    buf[1], buf[2] ) );
 
-    tls_version = mbedtls_ssl_read_version( buf + 1, ssl->conf->transport );
```

Comment:
Besides a pedantic check, the value is not otherwise used.  The rest of the fields are checked carefully, so at best, the specific check for 3 or 0xfe is of marginal additional value.

Reply:
ok